### DISCUSSION TITLE
Using gemmi. Simple case

### DISCUSSION THREAD

**Discussion by orex** [project-gemmi/gemmi#113] [Q&A] [answered]
Hello, Marcin.
I want you use your project in my program supercell (https://github.com/orex/supercell) for reading and storing cif files. At the moment  I'm using openbabel for this, but this is heavy library and I have a lot of problems with it (for example in CI/CD). I would like to have something lightweight. C++ header only library sounds perfectly fine.
I went through your documentation and code, but I can't found directly how to make a few simple steps. 
1) Read cif file.
2) Convert it to P1.
3) fract->cart
4) Do some processing.
5) Store unitcell (Eigen::Matrix3d), list of coordinates (+atom num, Label etc) to file.

The best answer for me will be is an example tool, which convert cif to supercell_cif.  

Thank you in advance. 

Best,
Kirill.

**Comment by wojdyr** [comment, marked as answer]
Is it for small molecule CIF?
1. see https://gemmi.readthedocs.io/en/latest/mol.html#small-molecules
2. I think expanding to P1 was considered, but there is no such function yet. There is a function to expand NCS (NCS – non-crystallgraphy statistics – is probably MX-specific).
3. Position pos = cell.orthogonalize(fractional);
4. n/a
5. Writing small molecule structures is not supported - it hasn't been a priority. Gemmi can write mmCIF files, PDB files and CIF files in a generic way. The latter could be used to write small molecule CIFs.

**Reply by orex** [reply]
Thank you very much for your quick response. I will check the options.
